internal/serialization/object_data.go

```diff
@@ -619,27 +614,18 @@ func (i *ObjectDataInput) readFloat64WithPosition(pos int32) (float64, error) {
 	return ret, err
 }
 
-func (i *ObjectDataInput) ReadUTF() string {
+func (i *ObjectDataInput) ReadString() string {
 	if i.err != nil {
 		return ""
 	}
-	var ret string
-	ret, i.err = i.readUTF()
-	return ret
-}
-
-func (i *ObjectDataInput) readUTF() (string, error) {
-	length, err := i.readInt32()
-	if err != nil || length == bufutil.NilArrayLength {
-		return "", err
-	}
-	var ret = make([]rune, length)
-	for j := 0; j < int(length); j++ {
-		r, n := utf8.DecodeRune(i.buffer[i.position:])
-		i.position += int32(n)
-		ret[j] = r
+	size, err := i.readInt32()
+	if err != nil || size == bufutil.NilArrayLength {
+		i.err = err
+		return ""
 	}
-	return string(ret), nil
+	s := string(i.buffer[i.position : i.position+size])
+	i.position += size
+	return s
 }
 
 func (i *ObjectDataInput) ReadUTFWithPosition(pos int32) string {
```

Comment:
nit: to make naming consistent with `ReadString`, it makes sense to rename this one to `ReadStringWithPosition`.